internal/auth/credentials.go

```diff
@@ -137,7 +137,7 @@ func (cred Credentials) IsTemp() bool {
 
 // IsServiceAccount - returns whether credential is a service account or not
 func (cred Credentials) IsServiceAccount() bool {
-	return cred.ParentUser != "" && (cred.Expiration.IsZero() || cred.Expiration.Equal(timeSentinel))
```

Comment:
IsTemp() also has a parent user, so IsServiceAccount() can return true for STS as well. We need to think for a better way. However I am not sure if this PR is really needed. STS accounts can work in all situations (MinIO SDKs, mc, AWS SDKs, ..)

Reply:
It was @abperiasamy requirement to avoid session tokens.

I am not sure why it's valuable yet but it is something we can definitely do. 

An account with parent user that doesn't have a session token is a service account.

Reply:
@Praveenrajmani you need fix the code to make sure this logic is true here

Reply:
pre-existing service account creds will have sessionToken set @harshavardhana 

all the credentials that we currently generate will have sessiontoken: https://github.com/minio/minio/blob/1b02e046c23a8547a01ea8e067d4ebdc0d02a1b3/internal/auth/credentials.go#L267

Reply:
> pre-existing service account creds will have sessionToken set @harshavardhana

yes, that is the issue. The session token contains the policy that will limit further the privileges of the service account.

Reply:
Well TBH even if session token is set, it can be decided to look at if it's a service account or not. We add explicit metadata that differ service accounts from STS